docs/technical-design.md

```diff
@@ -0,0 +1,130 @@
+}
+```
+
+The biggest challenge in deciding on an appropriate interval is that the time required to perform the reconciliation can vary a lot. Small changes may only require a 
+restart of the sidecar proxies or the Ingress gateway and are therefore much faster than a new installation or a Canary upgrade.  
+Given this dynamic in execution, we must always consider the slowest reconciliation process when defining `SyncPeriod`. Also, we have to consider that if we later add another controller to this operator, this new controller will also use the same frequency.  
+Using `RequeueAfter` on the other hand gives us more freedom, but also makes it a bit more complex to understand what is being done and why.
+
+We have decided to use `RequeueAfter` with a frequency of 5 minutes, as this gives us the ability to perform the matching as often as possible without the risk of repeatedly checking the slowest reconciliation process or piling up requests in the reconciliation queue.
+
+The queuing of reconciliation requests is handled by [controller-runtime](https://pkg.go.dev/sigs.k8s.io/controller-runtime) and is out of scope of this design.
+
+### Components
+#### IstioReconciler
+This is the reconciler that takes care of the entire Istio reconciliation process. 
+The responsibility of this reconciler is to fetch the current configuration and pass it on to the other reconciler together with the desired configuration. 
+It also controls the reconciliation process by running the reconcilers considering the execution dependencies between them.
+This reconciler should be run at least every 20 minutes.
+
+#### IstioInstallationReconciler
+This reconciler decides if an installation, upgrade or uninstall of Istio in the cluster must be done. The reconciler also creates the IstioOperator
+which is used to make changes to the Istio installation by passing it to the `IstioManager`.
+
+The installed IstioOperator is created by merging the `IstioCR` with the IstioOperator with Kyma default values.
+
+##### IstioManager
+This component contains the logic for managing the Istio installation. It knows about the supported client versions and forwards the 
+Istio API requests (e.g. Install, Upgrade) to the correct version of `IstioClient` if there are breaking changes.  
+Depending on how the Istio Go module behaves, it may not be necessary to support multiple clients for the time being. For the first simple implementation, we can probably do without the IstioManager for the time being.
+
+##### IstioClient
+A IstioClient encapsulates a specific version of the [Istio Go module](https://github.com/istio/istio). 
+As we want to support canary updates at some point we might need to support two version of the library if there are breaking changes.
+
+#### ProxySidecarReconciler
+This reconciler must be executed after the `IstioInstallationReconciler`. Its responsibility is to restart pods based on specific configuration changes.
```

Comment:
I think this might be partly answered by my reply [here](https://github.com/kyma-project/istio/pull/36#discussion_r1042104795) as we don't really want to have multiple controllers for simplicity.